src/App.tsx

```diff
@@ -1,8 +1,82 @@
 import React from 'react';
 import './App.css';
 
-const App = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+import { getGoods } from './api';
+import { GoodsList } from './components/GoodList/Goodlist';
+
+interface State {
+  goods: Good[];
+}
+
+class App extends React.Component<{}, State> {
+  state = {
+    goods: [],
+  };
+
+  handleAllGoods = () => {
+    getGoods()
+      .then(goods => {
+        this.setState({
+          goods,
+        });
+      });
+  };
+
+  handleClickFive = () => {
+    getGoods()
+      .then(goods => {
+        this.setState({
+          goods: goods
+            .sort((a: Good, b: Good) => a.name.localeCompare(b.name))
+            .slice(0, 5),
+        });
+      });
+  };
+
+  handleRedGoods = () => {
+    getGoods()
+      .then(goods => {
+        this.setState({
+          goods: goods.filter((good: Good) => good.color === 'red'),
+        });
+      });
+  };
+
+  render() {
+    const {
+      goods,
+    } = this.state;
```

Comment:
одно свойство можно в одну строку
```suggestion
    const { goods } = this.state;
```